src/lib/api/index.ts

```diff
@@ -101,78 +103,87 @@ export async function post(
     langs = opts.langs.slice(0, 3)
   }
 
-  let res
-  try {
-    opts.onStateChange?.(t`Posting...`)
-    res = await agent.post({
+  const rkey = TID.nextStr()
+  const uri = `at://${agent.assertDid}/app.bsky.feed.post/${rkey}`
+  const date = new Date().toISOString()
+
+  const writes: ComAtprotoRepoApplyWrites.Create[] = []
+
+  // Create post record
+  {
+    const record: AppBskyFeedPost.Record = {
+      createdAt: date,
```

Comment:
I believe the PDS will fill-in `$type: 'app.bsky.feed.post'` field on the record, but I think it may be good hygiene to do it explicitly here.